src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/conf/ReadPropertiesStepExecution.java

```diff
@@ -64,7 +64,9 @@
         if (!StringUtils.isBlank(step.getFile())) {
             FilePath f = ws.child(step.getFile());
             if (f.exists() && !f.isDirectory()) {
-                properties.load(f.read());
+                try(InputStream is = f.read()){
+                   properties.load(is); 
```

Comment:
@abayer is doing some refactoring in this code. I'll try to convince him to incorporate this change in that work as well, since it would be merge hell either way the order of how we merge this.

This would also need to bump the core dependency beyond 1.609.3 since that still technically is only java 6 and this is java 7 code. It's time to bump anyway ;)